pkg/messaging/brokers/brokers_nats.go

```diff
@@ -21,19 +23,21 @@ func init() {
 	log.Println("The binary was build using Nats as the message broker")
 }
 
-func NewPublisher(url string) (messaging.Publisher, error) {
+func NewPublisher(url string, tracer opentracing.Tracer) (messaging.Publisher, error) {
```

Comment:
Tracing should be added in `main.go`, not here. Tracing is a middleware just like any other.